CMakeLists.txt

```diff
@@ -0,0 +1,237 @@
+# folder in the search path for includes. So there this breaks.
+# As such, we 'fix' things here by also making sure the root folder is
+# included in the search path.
+if (LZO_FOUND)
+    get_filename_component(LZO_INCLUDE_DIRS "${LZO_INCLUDE_DIRS}/.." ABSOLUTE)
+endif (LZO_FOUND)
+
+include(AutodetectSSE REQUIRED)
+autodetect_sse()
+
+include(AutodetectXaudio2 REQUIRED)
+autodetect_xaudio2()
+
+include(AutodetectGrfcodec REQUIRED)
+autodetect_grfcodec()
+
+# IPO is only properly supported from CMake 3.9. Despite the fact we are
+# CMake 3.5, still enable IPO if we detect we are 3.9+.
+if (NOT CMAKE_VERSION VERSION_LESS 3.9)
+    cmake_policy(SET CMP0069 NEW)
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT IPO_FOUND)
+endif (NOT CMAKE_VERSION VERSION_LESS 3.9)
+
+show_options()
+
+if (UNIX AND NOT APPLE AND NOT OPTION_DEDICATED)
+    if (NOT SDL_FOUND)
+        message(FATAL_ERROR "SDL is required for this platform")
+    endif (NOT SDL_FOUND)
+endif (UNIX AND NOT APPLE AND NOT OPTION_DEDICATED)
+if (APPLE)
+    if (NOT AUDIOTOOLBOX_LIBRARY)
+        message(FATAL_ERROR "AudioToolbox is required for this platform")
+    endif (NOT AUDIOTOOLBOX_LIBRARY)
+    if (NOT AUDIOUNIT_LIBRARY)
+        message(FATAL_ERROR "AudioUnit is required for this platform")
+    endif (NOT AUDIOUNIT_LIBRARY)
+    if (NOT CARBON_LIBRARY)
+        message(FATAL_ERROR "Carbon is required for this platform")
+    endif (NOT CARBON_LIBRARY)
+    if (NOT COCOA_LIBRARY)
+        message(FATAL_ERROR "Cocoa is required for this platform")
+    endif (NOT COCOA_LIBRARY)
+endif (APPLE)
+
+if (MSVC)
+    # C++17 for MSVC
+    set(CMAKE_CXX_STANDARD 17)
+else (MSVC)
+    # C++11 for all other targets
+    set(CMAKE_CXX_STANDARD 11)
+endif (MSVC)
+
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_CXX_EXTENSIONS NO)
+
+list(APPEND GENERATED_SOURCE_FILES "${CMAKE_BINARY_DIR}/generated/rev.cpp")
+if (WIN32)
+    list(APPEND GENERATED_SOURCE_FILES "${CMAKE_BINARY_DIR}/generated/ottdres.rc")
+endif (WIN32)
+
+# Generate a target to determine version, which is execute every 'make' run
+add_custom_target(find_version
+        ${CMAKE_COMMAND}
+                -DFIND_VERSION_BINARY_DIR=${CMAKE_BINARY_DIR}/generated
+                -DCPACK_BINARY_DIR=${CMAKE_BINARY_DIR}
+                -P "${CMAKE_SOURCE_DIR}/cmake/scripts/FindVersion.cmake"
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        BYPRODUCTS ${GENERATED_SOURCE_FILES}
+)
+
+include(SourceList REQUIRED)
+include(Endian REQUIRED)
+add_endian_definition()
+
+# Needed by rev.cpp
+include_directories(${CMAKE_SOURCE_DIR}/src)
+# Needed by everything that uses Squirrel
+include_directories(${CMAKE_SOURCE_DIR}/src/3rdparty/squirrel/include)
+
+include(CompileFlags REQUIRED)
+compile_flags()
+
+add_executable(openttd WIN32 ${GENERATED_SOURCE_FILES})
+# All other files are added via target_sources()
+
+include(AddCustomXXXTimestamp REQUIRED)
+add_subdirectory(${CMAKE_SOURCE_DIR}/src)
```

Comment:
the directories are always relative to the current source directory, so this can be just "src".